plutus-core/plutus-ir/src/PlutusIR/Core/Instance/Pretty/Readable.hs

```diff
@@ -0,0 +1,145 @@
+  , PrettyReadableBy configName name
+  , GShow uni, Closed uni, uni `Everywhere` PrettyConst, Pretty fun
+  )
+
+instance PrettyConstraints configName tyname name uni fun
+          => PrettyBy (PrettyConfigReadable configName) (Term tyname name uni fun a) where
+    prettyBy = inContextM $ \case
+        Constant _ con         -> unitDocM $ pretty con
+        Builtin _ bi           -> unitDocM $ pretty bi
+        t | appLike t          ->
+          compoundDocM juxtFixity $ \ prettyIn ->
+            let ppArg (Left a)  = braces $ prettyIn ToTheRight botFixity a
+                ppArg (Right t) = prettyIn ToTheRight juxtFixity t
+            in prettyIn ToTheLeft juxtFixity fun <?> fillSep (map ppArg args)
+          where
+            appLike Apply{}  = True
+            appLike TyInst{} = True
+            appLike _        = False
+
+            (fun, args)                  = view t []
+            view (Apply _ fun arg)  args = view fun (Right arg:args)
+            view (TyInst _ fun arg) args = view fun (Left arg:args)
+            view fun                args = (fun, args)
+        Apply{} -> error "The impossible happened. Get a lottery ticket."
+        TyInst{} -> error "The impossible happened. Get a lottery ticket."
+        Var _ name             -> prettyM name
+        t@TyAbs{} ->
+            typeBinderDocM $ \prettyBinding prettyBody ->
+                ("/\\" <> align (fillSep (map (uncurry prettyBinding) args)) <+> "->") <?> prettyBody body
+            where (args, body)         = view t
+                  view (TyAbs _ n k b) = first ((n, k):) $ view b
+                  view t               = ([], t)
+        t@LamAbs{}  ->
+            compoundDocM binderFixity $ \prettyIn ->
+                let prettyBot x = prettyIn ToTheRight botFixity x
+                in ("\\" <> align (fillSep [ parens (prettyBot name <+> ":" <+> prettyBot ty)
+                                          | (name, ty) <- args ]) <+> "->") <?> prettyBot body
+            where (args, body)          = view t
+                  view (LamAbs _ n t b) = first ((n, t):) $ view b
+                  view t                = ([], t)
+        Unwrap _ term          ->
+            sequenceDocM ToTheRight juxtFixity $ \prettyEl ->
+                "unwrap" <+> prettyEl term
+        IWrap _ pat arg term   ->
+            sequenceDocM ToTheRight juxtFixity $ \prettyEl ->
+                "iwrap" <+> prettyEl pat <+> prettyEl arg <+> prettyEl term
+        Error _ ty             ->
+            compoundDocM juxtFixity $ \prettyIn ->
+                "error" <+> braces (prettyIn ToTheRight botFixity ty)
+        Let _ rec binds t         ->
+            compoundDocM binderFixity $ \prettyIn ->
+                let prettyBot x = prettyIn ToTheRight botFixity x
+                    prec | rec == NonRec = ""
+                         | otherwise     = "rec"
+                in align $ sep [ "let" <> prec <+> align (vcatHard (prettyBot <$> toList binds))
```

Comment:
See, I'm sure you thought hard about where to put all these `align`s and so on, but all that knowledge is about to be lost because you haven't written it down :)